content/essentials/signals.mdx

```diff
@@ -0,0 +1,116 @@
+```
+
+When a signal is called within a [tracking scope](/essentials/intro-to-reactivity#tracking), it forms a dependent between the signal and the wrapping function. 
+In this context, if the value of the signal changes, it notifies its dependencies of this change. 
+As a result, anything that is subscribed to this signal by accessing it within a tracking scope, will be called again. 
+This ensures that anything dependent on this signal will always be in sync with the state represented by these signals.
+
+<Aside>
+
+A tracking scope is typically created by [`createEffect`](/reference/createEffect)  or [`createMemo`](/reference/createMemo), other Solid primitives. 
+Both functions subscribe to the signals accessed within them, establishing a dependency relationship.
+
+When a signal within this scope changes its value, these dependencies are identified, and the computations (such as Solid's effects or memos) relying on them are triggered to update.
+
+</Aside>
+
+## Updating values
+
+To update a signal's value, you just need to call the setter function returned by `createSignal`:
+
+```jsx
+setCount(count()+1)
+
+console.log(count()) // output: 1
+```
+
+When the setter is called, the value of the signal is automatically updated, re-executing the functions.
+
+## Reacting to changes
+
+A distinct feature of signals is their ability to keep a list of their dependents. 
+[Effects](/essentials/effects) and [memos](/essentials/memos) subscribe to signals, inherently becoming _dependent _on them.
+This ensures tracking across all the dependent functions whenever the signal's value changes.
+
+```jsx
+function Counter() {
+  const [count, setCount] = createSignal(0);
+  
+  // logs initial value of count
+  console.log(count()) // result: 0
+  
+  // creates a reactive value `double` that is always twice the value of `count`
+  const double = createMemo(() => count() * 2);
+  
+  return (
+    <div>
+      <p>Count: {count()}</p>
+      <p>Double: {double()}</p>
+      <button onClick={() => setCount(count() + 1)}>Increment</button>
+    </div>
+  );
+}
+
+export default Counter;
+```
+In this example, `createSignal` initializes a reactive state variable `count` and sets its initial value to `0` .
+
+In the top-level code, including the `callonsole.log` statement, execution happens only once—when the component is initially rendered.
+Components in solid _are not_ re-executed to update.
+This means that the initial value of `count`, which is `0`, will be logged to the console only once, regardless of subsequent changes to `count`.
+
+On the other hand, `double` is initialized using `createMemo`, another Solid primitive, which forms a reactive computed value based on `count` signal.
+Unlike the `console.log` statement, the `createMemo` function ensures that `double` tracks the changes in `count`. 
+This implies that `double` automatically recalculates and prompts the component to re-render, displaying the updated value whenever `count` is updated.
```

Comment:
Solid only re-renders where the reactive values are used. It doesn't necessarily re-render the component, only the html tag/tags that are dependent on that value